docs/tutorials/domain_tutorial.jl

```diff
@@ -0,0 +1,140 @@
+#=
+# Domain Tutorial
+
+## Goals of the tutorial
+The goal of this is to outline what is currently implemented in ClimaLSM 
+and to serve as a software design document 
+for future development involving the underlying domains.
+
+## Background
+
+In both the atmosphere and the ocean, all variables are defined at all locations
+in the region of interest, or domain.  For example, the air density, temperature, pressure,
+and wind speed are defined everywhere in the domain. After choosing a resolution
+and discretizing space, the numerical problem is to advance a 
+system of ordinary differential
+equations, where at each coordinate point a value of 
+`ρ`, `T`, `P`, and `u⃗` are solved for at each step. The choice of domain is a question "only"
+of geometry: you may be interested in a large eddy simulation (using a box domain), or 
+in a global model (where you would need a spherical shell domain
+representing the atmosphere or ocean from some depth to a given height).
+
+For land surface
+models, each variable is not defined everywhere in space. For example,
+the soil water content `θ` is only defined below ground. Snow water equivalent (`S`) is only
+defined on the surface itself. Canopy variables are only defined above ground.
+ Once we have discretized the land surface region into
+a set of points, the numerical problem is to advance a system of ODEs, where
+at each coordinate point a different subset of (`θ`, `S`, ...) are solved for.
+
+In other words, different prognostic variables in land surface models exist
+in different, overlapping, domains. We need to decide on the geometry of interest (e.g. single column
+vs a global simulation), but we also need to specify where each variable of the model is defined.
+
+ClimaLSM Domains were designed with this in mind. Following other Clima models, the domains are defined
+so that (1) the user can easily switch geometries, e.g. single column to global model, (2) individual component
+models can be run by themselves, using a single domain, and so that (3) different model component domains
+ can be combined into a single land model domain, since different land model components can be combined
+to run as a single land surface model.
```

Comment:
I think a list here would be nice
```suggestion
so that 
1.  the user can easily switch geometries, e.g. single column to global model,
2.  individual component models can be run by themselves, using a single domain,
3.  different model component domains can be combined into a single land model domain, 
since different land model components can be combined to run as a single land surface model.
```